server/re/src/irods_re_serialization.cpp

```diff
@@ -1199,6 +1200,94 @@ namespace irods::re_serialization
+            _out["inpParamArray_oprType"] = boost::lexical_cast<std::string>(msparamarray_ptr->oprType);
+            for (int i = 0; i < msparamarray_ptr->len; i++) {
+                std::string prefix = fmt::format("inpParamArray_{}_", i);
+                _out[prefix + "label"] = msparamarray_ptr->msParam[i]->label;
+                _out[prefix + "type"] = msparamarray_ptr->msParam[i]->type;
+                const auto* msparam_ptr = msparamarray_ptr->msParam[i];
+
+                if (msparam_ptr == nullptr || msparam_ptr->inOutStruct == nullptr) {
+                    _out[prefix + "inOutStruct"] = "nullptr";
+                }
+                else if (strcmp(msparam_ptr->type, STR_MS_T) == 0) {
+                    _out[prefix + "inOutStruct"] = static_cast<char*>(msparam_ptr->inOutStruct);
+                }
+                else if (strcmp(msparam_ptr->type, INT_MS_T) == 0) {
+                    _out[prefix + "inOutStruct"] = std::to_string(parseMspForPosInt(const_cast<MsParam*>(msparam_ptr)));
+                }
+                else if (strcmp(msparam_ptr->type, DOUBLE_MS_T) == 0) {
+                    double out = 0.0;
+                    if (parseMspForDouble(const_cast<MsParam*>(msparam_ptr), &out)) {
+                        THROW(SYS_INVALID_INPUT_PARAM, "Failed to convert microservice argument to float.");
+                    }
+                    _out[prefix + "inOutStruct"] = std::to_string(out);
+                }
+                else if (strcmp(msparam_ptr->type, FLOAT_MS_T) == 0) {
+                    float out = 0.0;
+                    if (parseMspForFloat(const_cast<MsParam*>(msparam_ptr), &out)) {
+                        THROW(SYS_INVALID_INPUT_PARAM, "Failed to convert microservice argument to float.");
+                    }
+                    _out[prefix + "inOutStruct"] = std::to_string(out);
+                }
+                else {
+                    _out[prefix + "inOutStruct"] = fmt::format("unrepresentable_type_{}", msparam_ptr->type);
+                }
+                const auto* ptr = msparamarray_ptr->msParam[i]->inpOutBuf;
+                if (ptr) {
+                    _out[prefix + "inpOutBuf_len"] = boost::lexical_cast<std::string>(ptr->len);
+                    std::string& str = _out[prefix + "inpOutBuf_buf"];
+
+                    for (int j = 0; j < ptr->len; j++) {
+                        unsigned char c = *(static_cast<unsigned char*>(ptr->buf) + j);
+
+                        if (isprint(c) || isspace(c) || isblank(c)) {
+                            str += c;
+                        }
+                        else {
+                            str += fmt::format("\\x{0:02x}", c);
+                        }
+                    }
+                }
+                else {
+                    _out[prefix + "null_value"] = "null_value";
+                }
+            }
+            return SUCCESS();
+        }
+        catch (const boost::bad_any_cast& e) {
```

Comment:
Needs a catch block for `const irods::exception&` due to the use of `THROW(...)` above.

Use `e.code()` to get the iRODS error code from the exception.